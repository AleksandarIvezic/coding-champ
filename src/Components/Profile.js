import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import mockUser from '../Library/mockUser';
import '../Styles/Profile.css';

const Profile = () => {
  let url;
  if (mockUser.avatar) url = mockUser.avatar;
  else url = 'programmer.png';
  return (
    <>
      <header className="profile-header">
        <img src={url} alt="avatar" />
        <div className="info">
          <h3>
            Name:
            <span>{` ${mockUser.name}`}</span>
          </h3>
          <h3>
            E-mail:
            <span>{` ${mockUser.email}`}</span>
          </h3>
        </div>
      </header>
      <div className="best-scores">
        <h3>Best scores</h3>
        <table>
          <thead>
            <tr>
              <th>Categories</th>
              <th>Score</th>
            </tr>
          </thead>
          <tbody>
            {mockUser.bestScores.map((category) => (
              <tr key={uuidv4()}>
                <td>{Object.keys(category)[0]}</td>
                <td>{category[Object.keys(category)[0]]}</td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
      <div className="trophies">
        <h3>Trophies</h3>
        {mockUser.trophies.map((trophy) => (
          <h4 key={uuidv4()}>{trophy}</h4>
        ))}
      </div>
    </>
  );
};

export default Profile;
