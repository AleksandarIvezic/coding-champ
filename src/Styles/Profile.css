.profile-header,
.best-scores {
  display: flex;
  justify-content: space-evenly;
  margin: 50px 140px;
}
