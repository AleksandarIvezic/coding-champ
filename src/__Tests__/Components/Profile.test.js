import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '../../Components/Profile';

it('renders correctly', () => {
  const tree = renderer
    .create(<Profile />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
