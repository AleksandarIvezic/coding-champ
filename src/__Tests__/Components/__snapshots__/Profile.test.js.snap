// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="profile-header"
  >
    <img
      alt="avatar"
      src="programmer.png"
    />
    <div
      className="info"
    >
      <h3>
        Name:
        <span>
           User1
        </span>
      </h3>
      <h3>
        E-mail:
        <span>
           user@mail.com
        </span>
      </h3>
    </div>
  </header>,
  <div
    className="best-scores"
  >
    <h3>
      Best scores
    </h3>
    <table>
      <thead>
        <tr>
          <th>
            Categories
          </th>
          <th>
            Score
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            HTML
          </td>
          <td>
            124322
          </td>
        </tr>
        <tr>
          <td>
            CSS
          </td>
          <td>
            124452
          </td>
        </tr>
        <tr>
          <td>
            JavaScript
          </td>
          <td>
            1243362
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="trophies"
  >
    <h3>
      Trophies
    </h3>
    <h4>
      bone-marrow
    </h4>
    <h4>
      make-up artist
    </h4>
    <h4>
      conditional-Trainer
    </h4>
  </div>,
]
`;
